src/mscorlib/src/System/IO/StreamHelpers.CopyValidation.cs

```diff
@@ -20,25 +20,23 @@ public static void ValidateCopyToArgs(Stream source, Stream destination, int buf
                 throw new ArgumentOutOfRangeException(nameof(bufferSize), bufferSize, Environment.GetResourceString("ArgumentOutOfRange_NeedPosNum"));
             }
 
-            bool sourceCanRead = source.CanRead;
-            if (!sourceCanRead && !source.CanWrite)
+            if (!source.CanRead)
             {
-                throw new ObjectDisposedException(null, Environment.GetResourceString("ObjectDisposed_StreamClosed"));
-            }
-
-            bool destinationCanWrite = destination.CanWrite;
-            if (!destination.CanRead && !destinationCanWrite)
-            {
-                throw new ObjectDisposedException(nameof(destination), Environment.GetResourceString("ObjectDisposed_StreamClosed"));
-            }
+                if (!source.CanWrite)
+                {
+                    throw new ObjectDisposedException(null, Environment.GetResourceString("ObjectDisposed_StreamClosed"));
```

Comment:
I know it was like this before, but it seems inconsistent that we're passing in `null` to the ODE here and `nameof(destination)` to the ODE below. Maybe this should be changed to `nameof(source)`.